cumulusci/tasks/salesforce/EnsureRecordTypes.py

```diff
@@ -0,0 +1,140 @@
+            "required": True,
+        },
+    }
+    _deploy = Deploy
+
+    def _init_options(self, kwargs):
+        super(EnsureRecordTypes, self)._init_options(kwargs)
+
+        self.options["generate_business_process"] = False
+
+        # Validate developer name
+        if not re.match(r"^\w+$", self.options["record_type_developer_name"]):
+            raise TaskOptionsError(
+                "Record Type Developer Name value must contain only alphanumeric or underscore characters"
+            )
+
+    def _infer_business_process(self):
+        # If our sObject is Lead or Opportunity, we need to generate businessProcess
+        # metadata to make the record type deployable.
+        sobject = self.options["sobject"]
+
+        if sobject in SOBJECT_MAP:
+            self.options["generate_business_process"] = True
+            describe_results = getattr(self.sf, sobject).describe()
+            # Salesforce requires that at least one picklist value be present and active
+            self.options["stage_name"] = list(
+                filter(
+                    lambda pl: pl["active"],
+                    list(
+                        filter(
+                            lambda f: f["name"] == SOBJECT_MAP[sobject],
+                            describe_results["fields"],
+                        )
+                    )[0]["picklistValues"],
+                )
+            )[0]["value"]
+
+    def _build_package(self):
+        objects_app_path = "objects"
+        os.mkdir(objects_app_path)
+        with open(
+            os.path.join(objects_app_path, self.options["sobject"] + ".object"), "w"
+        ) as f:
+            if self.options["generate_business_process"]:
+                business_process_metadata = BUSINESS_PROCESS_METADATA.format(
+                    record_type_developer_name=self.options[
+                        "record_type_developer_name"
+                    ],
+                    stage_name=self.options["stage_name"],
+                )
+                business_process_link = BUSINESS_PROCESS_LINK.format(
+                    record_type_developer_name=self.options[
+                        "record_type_developer_name"
+                    ]
+                )
+            else:
+                business_process_metadata = business_process_link = ""
+
+            f.write(
+                SOBJECT_METADATA.format(
+                    record_type_developer_name=self.options[
+                        "record_type_developer_name"
+                    ],
+                    record_type_label=self.options["record_type_label"],
+                    business_process_metadata=business_process_metadata,
+                    business_process_link=business_process_link,
+                )
+            )
+        with open("package.xml", "w") as f:
+            f.write(PACKAGE_XML)
+
+    def _run_task(self):
+        self._infer_business_process()
```

Comment:
At least for the NPSP use case, part of it is that we don't want to create the NPSP Default record type if there are already opportunity record types. We also don't want to overwrite the record type metadata if it already exists. So I think we need to start by retrieving the specified sobject to see what record types it already has.

Reply:
Agreed. Resolving this in a new commit.